CMakeLists.txt

```diff
@@ -1,29 +1,26 @@
 ﻿cmake_minimum_required (VERSION 3.10)
 
-if (DEFINED BLT_SOURCE_DIR)
-    # Support having a shared BLT outside of the repository if given a BLT_SOURCE_DIR
-    if (NOT EXISTS ${BLT_SOURCE_DIR}/SetupBLT.cmake)
-        message(FATAL_ERROR "Given BLT_SOURCE_DIR does not contain SetupBLT.cmake")
-    endif()
-else()
-    # Use internal BLT if no BLT_SOURCE_DIR is given
-    set(BLT_SOURCE_DIR "${PROJECT_SOURCE_DIR}/external/blt" CACHE PATH "")
-    if (NOT EXISTS ${BLT_SOURCE_DIR}/SetupBLT.cmake)
-        message(FATAL_ERROR
-            "The BLT git submodule is not present. "
-            "Either run the following two commands in your git repository: \n"
-            "    git submodule init\n"
-            "    git submodule update\n"
-            "Or add -DBLT_SOURCE_DIR=/path/to/blt to your CMake command." )
-    endif()
-endif()
 
 # Default to C++14 if not set TODO: check to ensure this is the minimum
 if (NOT BLT_CXX_STD)
     set(BLT_CXX_STD "c++14" CACHE STRING "")
 endif()
 
-include(external/blt/SetupBLT.cmake)
+include(cmake/blt_boilerplate.cmake)
+include(cmake/macros.cmake)
+
+# list of backends to add as dependencies
+if(ENABLE_OPENMP)
```

Comment:
That's something we should discuss.  BLT uses unprefixed versions of all of these, and I'm checking for the names it uses (except the HIP one I'm fixing).  Mainly that's to keep it project independent and make multiple nested BLT projects work with the same options, so say add_subdirectory is well-behaved.

I'm perfectly good with adding in `DESUL_*` versions of all of these that then feed into the generic ones if that helps, but it would be pretty invasive to get rid of the unprefixed ones completely.